### ISSUE TITLE
Once you've edited the project file once you can no longer modify the project

### ISSUE THREAD

**Issue by davkean** [dotnet/roslyn-project-system#812]
1. Right-click on a project, choose Edit "ProjectFile"
2. Close editor
3. Add a new folder to the project

Expected: For this to succeed
Actual:

```
Microsoft Visual Studio
---------------------------
Could not find a part of the path 'C:\Users\davkean\AppData\Local\Temp\tmpB4B7.tmp\ConsoleApp8.csproj'.
---------------------------
OK   
---------------------------
```
Stack:

```
 	mscorlib.dll!System.IO.__Error.WinIOError(int errorCode, string maybeFullPath)	Unknown
 	mscorlib.dll!System.IO.FileStream.Init(string path, System.IO.FileMode mode, System.IO.FileAccess access, int rights, bool useRights, System.IO.FileShare share, int bufferSize, System.IO.FileOptions options, Microsoft.Win32.Win32Native.SECURITY_ATTRIBUTES secAttrs, string msgPath, bool bFromProxy, bool useLongPath, bool checkHost)	Unknown
 	mscorlib.dll!System.IO.FileStream.FileStream(string path, System.IO.FileMode mode, System.IO.FileAccess access, System.IO.FileShare share, int bufferSize, System.IO.FileOptions options, string msgPath, bool bFromProxy, bool useLongPath, bool checkHost)	Unknown
 	mscorlib.dll!System.IO.StreamWriter.CreateFile(string path, bool append, bool checkHost)	Unknown
 	mscorlib.dll!System.IO.StreamWriter.StreamWriter(string path, bool append, System.Text.Encoding encoding, int bufferSize, bool checkHost)	Unknown
 	mscorlib.dll!System.IO.File.InternalWriteAllText(string path, string contents, System.Text.Encoding encoding, bool checkHost)	Unknown
 	mscorlib.dll!System.IO.File.WriteAllText(string path, string contents)	Unknown
>Microsoft.VisualStudio.ProjectSystem.Managed.dll!Microsoft.VisualStudio.ProjectSystem.Utilities.FileSystem.WriteAllText(string filePath, string content) Line 48	C#
 Microsoft.VisualStudio.ProjectSystem.Managed.VS.dll!Microsoft.VisualStudio.ProjectSystem.VS.Editor.MsBuildModelWatcher.XmlHandler(string xml) Line 49	C#
 Microsoft.VisualStudio.ProjectSystem.Managed.VS.dll!Microsoft.VisualStudio.ProjectSystem.VS.Editor.MsBuildModelWatcher.ProjectXmlHandler(object sender, Microsoft.Build.Evaluation.ProjectXmlChangedEventArgs args) Line 40	C#
 Microsoft.Build.dll!Microsoft.Build.Evaluation.ProjectCollection.OnProjectXmlChanged(Microsoft.Build.Evaluation.ProjectXmlChangedEventArgs e)	Unknown
 Microsoft.Build.dll!Microsoft.Build.Evaluation.ProjectCollection.ProjectRootElementCache_ProjectRootElementDirtiedHandler(object sender, Microsoft.Build.Evaluation.ProjectXmlChangedEventArgs e)	Unknown
 Microsoft.Build.dll!Microsoft.Build.Evaluation.ProjectRootElementCache.OnProjectRootElementDirtied(Microsoft.Build.Construction.ProjectRootElement sender, Microsoft.Build.Evaluation.ProjectXmlChangedEventArgs e)	Unknown
 	Microsoft.Build.dll!Microsoft.Build.Construction.ProjectRootElement.MarkDirty(string reason, string param)	Unknown
 	Microsoft.Build.dll!Microsoft.Build.Construction.ProjectElement.MarkDirty(string reason, string param)	Unknown
 Microsoft.Build.dll!Microsoft.Build.Construction.ProjectElementContainer.AddInitialChild(Microsoft.Build.Construction.ProjectElement child)	Unknown
 Microsoft.Build.dll!Microsoft.Build.Construction.ProjectElementContainer.PrependChild(Microsoft.Build.Construction.ProjectElement child)	Unknown
 Microsoft.Build.dll!Microsoft.Build.Construction.ProjectElementContainer.InsertAfterChild(Microsoft.Build.Construction.ProjectElement child, Microsoft.Build.Construction.ProjectElement reference)	Unknown
 	Microsoft.Build.dll!Microsoft.Build.Construction.ProjectItemGroupElement.AddItem(string itemType, string include, System.Collections.Generic.IEnumerable<System.Collections.Generic.KeyValuePair<string, string>> metadata)	Unknown
 	Microsoft.Build.dll!Microsoft.Build.Construction.ProjectRootElement.AddItem(string itemType, string include, System.Collections.Generic.IEnumerable<System.Collections.Generic.KeyValuePair<string, string>> metadata)	Unknown
 	Microsoft.Build.dll!Microsoft.Build.Evaluation.Project.AddItem(string itemType, string unevaluatedInclude, System.Collections.Generic.IEnumerable<System.Collections.Generic.KeyValuePair<string, string>> metadata)	Unknown
 Microsoft.VisualStudio.ProjectSystem.dll!Microsoft.VisualStudio.ProjectSystem.ProjectItemProviderBase<Microsoft.VisualStudio.ProjectSystem.IProjectItem>.AddItemsToProjectAsync(Microsoft.VisualStudio.ProjectSystem.ProjectWriteLockReleaser access, Microsoft.Build.Evaluation.Project project, System.Collections.Generic.IEnumerable<System.Tuple<string, string, System.Collections.Generic.IEnumerable<System.Collections.Generic.KeyValuePair<string, string>>>> items)	Unknown
 mscorlib.dll!System.Runtime.CompilerServices.AsyncTaskMethodBuilder<System.Collections.Generic.ICollection<Microsoft.VisualStudio.ProjectSystem.IProjectItem>>.Start<Microsoft.VisualStudio.ProjectSystem.ProjectItemProviderBase<Microsoft.VisualStudio.ProjectSystem.IProjectItem>.<AddItemsToProjectAsync>d__96>(ref Microsoft.VisualStudio.ProjectSystem.ProjectItemProviderBase<Microsoft.VisualStudio.ProjectSystem.IProjectItem>.<AddItemsToProjectAsync>d__96 stateMachine)	Unknown
 Microsoft.VisualStudio.ProjectSystem.dll!Microsoft.VisualStudio.ProjectSystem.ProjectItemProviderBase<Microsoft.VisualStudio.ProjectSystem.IProjectItem>.AddItemsToProjectAsync(Microsoft.VisualStudio.ProjectSystem.ProjectWriteLockReleaser access, Microsoft.Build.Evaluation.Project project, System.Collections.Generic.IEnumerable<System.Tuple<string, string, System.Collections.Generic.IEnumerable<System.Collections.Generic.KeyValuePair<string, string>>>> items)	Unknown
Microsoft.VisualStudio.ProjectSystem.dll!Microsoft.VisualStudio.ProjectSystem.ProjectItemProviderBase<Microsoft.VisualStudio.ProjectSystem.IProjectItem>.AddAsync(System.Collections.Generic.IEnumerable<System.Tuple<string, string, System.Collections.Generic.IEnumerable<System.Collections.Generic.KeyValuePair<string, string>>>> items)	Unknown
 ```

**Comment by 333fred** [comment]
Same root cause as #787.

**Comment by srivatsn** [comment]
Fixed by #835
